packages/itwin/tree-widget/src/test/trees/models-tree/UseModelsTree.test.tsx

```diff
@@ -0,0 +1,253 @@
+          const element1 = insertPhysicalElement({ builder, codeValue: "element1", categoryId: category1, modelId: physicalModel1, userLabel: "Element1" }).id;
+          const element2 = insertPhysicalElement({ builder, codeValue: "element2", categoryId: category2, modelId: physicalModel2, userLabel: "Element2" }).id;
+          const element3 = insertPhysicalElement({ builder, codeValue: "element3", categoryId: category3, modelId: physicalModel3, userLabel: "Element3" }).id;
+          return {
+            models: [physicalModel1, physicalModel2, physicalModel3],
+            categories: [category1, category2, category3],
+            elements: [element1, element2, element3],
+          };
+        });
+      }
+      before(async function () {
+        await initializePresentationTesting({
+          backendProps: {
+            caching: {
+              hierarchies: {
+                mode: HierarchyCacheMode.Memory,
+              },
+            },
+          },
+          rpcs: [IModelReadRpcInterface, PresentationRpcInterface, ECSchemaRpcInterface],
+        });
+        // eslint-disable-next-line @itwin/no-internal
+        ECSchemaRpcImpl.register();
+        await TreeWidget.initialize();
+        const buildIModelResult = await createIModel(this);
+        imodel = buildIModelResult.imodel;
+        categoryIds = buildIModelResult.categories;
+        modelIds = buildIModelResult.models;
+        elementIds = buildIModelResult.elements;
+        viewport = OffScreenViewport.create({
+          view: await createViewState(imodel, categoryIds, modelIds),
+          viewRect: new ViewRect(),
+        });
+        initialProps = { activeView: viewport };
+        imodelAccess = createIModelAccess(imodel);
+        abortSignal = new AbortController().signal;
+        getSubTreePaths = async ({ createInstanceKeyPaths }) => {
+          return createInstanceKeyPaths({
+            targetItems: [
+              { className: elementClass, id: elementIds[0] },
+              { className: categoryClass, id: categoryIds[1] },
+            ],
+          });
+        };
+        selectionStorage = createStorage();
+      });
+
+      beforeEach(() => {
+        selectionStorage.clearStorage({ imodelKey: imodel.key });
+      });
+
+      after(async function () {
+        viewport[Symbol.dispose]();
+        await imodel.close();
+        await terminatePresentationTesting();
+        TreeWidget.terminate();
+      });
+
+      it("getFilteredPaths returns correct result when getSubTreePaths is not defined", async () => {
+        const { result: renderHookResult } = renderHook(useModelsTree, { initialProps });
+        const { getFilteredPaths } = renderHookResult.current.modelsTreeProps;
+        await waitFor(async () => {
+          const result = getFilteredPaths ? await getFilteredPaths({ imodelAccess, abortSignal }) : getFilteredPaths;
```

Comment:
I don't understand this. 
1. `getFilteredPaths` prop is an optional function, and its result is `Promise<HierarchyFilteringPath[]>`. What's the point to assign `result` either result of `getFilteredPaths`, or `getFilteredPaths` itself?
2. Why do we even need this kind of check in the test? Don't we know what we expect?